src/index.cpp

```diff
@@ -2939,9 +2714,12 @@ int Index<T, TagT, LabelT>::insert_point(const T *point, const TagT tag, const s
                 }
 
                 auto fz_location = (int)(_max_points) + _frozen_pts_used; // as first _fz_point
-                _labels.insert(label);
-                _label_to_start_id[label] = (uint32_t)fz_location;
-                _location_to_labels[fz_location] = {label};
+                _filter_store->add_to_label_set(label);
+                _filter_store->update_medoid_by_label(label, (uint32_t)fz_location);
+                std::vector<LabelT> fz_label = {label};
```

Comment:
Can we have single method in filter_store like update_labels and in implementation we can use add_to_label_set, update_medoid_by_label and set_labels_to_location

Reply:
Let me get back to you on this.